packages/common/src/EthersSignerUtils.ts

```diff
@@ -16,9 +16,10 @@ function getPrivateKeySigner() {
   return new Wallet(process.env.PRIVATE_KEY);
 }
 
-export async function getGckmsSigner() {
-  if (!args.keys) throw new Error(`Wallet GCKSM selected but no keys parameter set! Set GCKMS key to use`);
-  const privateKeys = await retrieveGckmsKeys(getGckmsConfig([args.keys]));
+export async function getGckmsSigner(selectedWallet?: string): Promise<Wallet> {
+  if (!args.keys && !selectedWallet)
+    throw new Error(`Wallet GCKSM selected but no keys parameter set or selectedWallet! Set any of them first`);
+  const privateKeys = await retrieveGckmsKeys(getGckmsConfig([args.keys ?? selectedWallet]));
```

Comment:
How is selectedWallet intended to be provided? Any reason not to pull it from an env so the script doesn't have to worry about figuring it out? Or do you think scripts will have hardcoded names for the keys they want?

Reply:
I think the point of this it to let you use: a) GCKMs provide and b) hardhat at the same time and pass it in _from the script_ when calling this method. what you're proposing Matt is a special env variable that we can use to set what wallet you want. one downside with doing it this way is that you then need a special env to run these hardat + gckms scripts, which is not ideal, vs just the script being able to select (which itself is not ideal). I think optimally we change scripts to _not_ need hardhat to run (i.e they are node scripts) and then we can use this provider logic in the way it was originally intended but that's not always the case.

another option could be to enable _both_ (i.e pass it in from the caller or an env or an argument).

Reply:
I think the overarching point that GCKMS + hardhat is pretty clunky. +1.

The only reason I'm pushing on this is that I think most people using these scripts are used to having the wallet determined by the environment + CLI args, and not the script, itself. I think it's mainly a UX consistency thing to have the wallet be something that is fully determined outside the script.

Side note: the GCKMS config already has to be passed in via the env.

Side note side note: I would love to find a better/simpler way to configure GCKMS, as well -- the JSON env is super clunky. There may be opportunities to make that cleaner/simpler with more aggressive defaulting.